pkg/labeller/labeller.go

```diff
@@ -18,6 +18,7 @@ type FetchableObject struct {
 type Labeller struct {
 	APIProxy       apiproxy.APIProxy
 	ObjectMeta     *metav1.ObjectMeta
+	ComponentType  string
```

Comment:
It is rather confusing that you are adding ComponentType field and it doesn't have `consts.ComponentType` type and content of those consts.

Maybe we could reduce complexity instead of adding it? 
I think it would be simpler if Labeller would look something like:
```
type Labeller struct {
	apiProxy       apiproxy.APIProxy
	objectMeta     *metav1.ObjectMeta
        componentType const.ComponentType
        componentNamePart string // role for hp, name for others, may be empty
        annotations    map[string]string
}
```
and some methods over those fields to get full label and full name.
internal map const.ComponentType <-> YTComponentLabel* may be introduced to reduce possible errors and simplify labeller usage.

I would prefer to have some tests with canonized labels here before refactoring, though I'm not sure if it is easy to write since labeller is configured inside those components and currently the easiest way looks like to fetch such labels from k8s resources after applying.

Reply:
Yeah, the annoying part is that it is an even bigger refactoring, since ComponentType and ComponentLabel are almost the same (except for CamelCase vs lower-case-hyphenated), but slightly different. 

I guess the best way for now is indeed to have an internal function which maps these types 1:1 without changing the names.

Reply:
As for tests, maybe I can try fetching these labels in an e2e test. Not sure I want to do this for all components though.

Reply:
> Yeah, the annoying part is that it is an even bigger refactoring

I see it something like:
- `ComponentLabel` usage are auto-replaced with a new `ComponentLabel()` call
- `ComponentName` usages are replaced with a new `FullComponentName()` call
- Labeller initialization is replaced with `NewLabeller(..., componentType ComponentType, componentNamePart string //may be empty)`
- `ComponentType()` and other methods for labels are added.
Do you see more things to be done?

Reply:
Implemented this ~as you said. Had to get rid of `DeprecatdeChytController` for simplicity, but I think that's fine.